src/python/pants/backend/python/target_types.py

```diff
@@ -393,6 +393,8 @@ class PexEnvField(DictStringToStringField):
 
 class PexPlatformsField(StringSequenceField):
     alias = "platforms"
+    removal_version = "2.20.0.dev0"
```

Comment:
I imagine this may be quite impactful across the pants ecosystem, and I believe the main motivation for this is that it is finicky to work with and easy to give the wrong result, rather than significant Pants-internal code simplification?

As such, maybe we could have a longer deprecation period to ease user churn?

```suggestion
    removal_version = "2.23.0.dev0"
```
(Made up, but approximately 6 months, with 6 week releases?)

Reply:
True, this doesn't lead to simplification in Pants, but it does lead to simplification of our lives. Since `platforms` is broken, it is a bit of a landmine in the user's codebase, and when it breaks they first turn to Slack, which costs us time and support resources. So we do want to shovel them off it ASAP. 

I could go for 2.22?

Reply:
Yeah, that makes sense. Happy with 2.22! (Given the warning isn't easily disablable, I would _imagine_ that most people switch away in 2.19...)